### ISSUE TITLE
[Feat] Use Ubuntu 24.04 runners in CI

### ISSUE THREAD

**Issue by echoix** [OSGeo/grass#3794]
**Is your feature request related to a problem? Please describe.**
For 8.5.0, we should prepare to use the Ubuntu 24.04 runners in CI. A little less software is installed by default, they reduced the default enabled repositories, so we would need to make sure the list of software to install is up to date. Maybe we would have some issues with some extra software not available as is, so maybe we would need to check if the ubuntugis ppa fits our needs (and we could see if the builds work with newest and latest of dependencies).

**Comment by echoix** [comment]
I just saw the announcement posted 5 days ago in actions/runner-images: Ubuntu-latest workflows will use Ubuntu-24.04 image

- [ ] https://github.com/actions/runner-images/issues/10636

Starting from September 23, 2024 and expect to be completed October 30th, 2024.

We have some jobs using ubuntu-latest:
![image](https://github.com/user-attachments/assets/315d7b37-b485-4779-ab4d-140b4841a990)

**Comment by wenzeslaus** [comment]
Do you have an idea how to test the ubuntu-latest workflows with new version besides switching them to a specific version? Is that want we want? Seems more straightforward for transitions.

**Comment by echoix** [comment]
> Do you have an idea how to test the ubuntu-latest workflows with new version besides switching them to a specific version? Is that want we want? Seems more straightforward for transitions.

- [x] Clang format:
  - runs a docker action, so doesn't matter there,
  - plus the create upload suggestion (ours) that is some bash lines (so doesn't matter)
  - upload artifact (node) (so doesn't matter)
- [ ] Create release draft:
  - uses Python, and correctly uses setup-Python (we might want to upgrade it)
  - Might need to check if ./utils/generate_version_git_file.sh assumes that some tools that were available are still available
  - softprops/action-gh-release is a node action, so doesn't matter
- [ ] Docker:
  - the build push actions and all others should be working fine.
- [x] Labeler: is a node action, so doesn't matter
- [ ] periodic update:
  - checkout: node action, doesn't matter
  - git: still there
  - wget: should still be there
  - peter-evans/create-pull-request: node action, doesn't matter
- [ ] Post pr reviews:
  - download artifact: node action
  - reviewdog/action-setup + installed reviewdog: reviewdog is in go. Should be statically compiled, shouldn't be having problems. Unknown, but surprising if it doesn't work.
  - bash code: doesn't really matter
- [x] super linter:
  - checkout: node action
  - super-linter: is a docker action, so it will just run the container
- [ ] titles:
  - checkout: node action, doesn't matter
  - we run a python script without installing a python version with setup-Python: if ever our code doesn't work anymore with the future Python versions, we should install a specific one (taking more time). We will get the latest of 3.12.
- [x] Verify success:
  - bash only logic

**Comment by wenzeslaus** [comment]
So that looks like we are in good shape, right?

**Comment by echoix** [comment]
Yep. The more touchy ones are specified as 22.04. We should still be preparing our build scripts/dependency lists for 24.04. No one had time since June to do it yet.

**Comment by thura784** [comment]
src/features/marketplace/components/TradeableList.tsx

**Comment by ninsbl** [comment]
Is this still an issue? We are on Ubuntu 24.04 in CI, no? Anything left here?
